pulsar-client/src/test/java/org/apache/pulsar/client/tutorial/SampleCryptoProducer.java

```diff
@@ -0,0 +1,76 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.client.tutorial;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.pulsar.client.api.CryptoKeyReader;
+import org.apache.pulsar.client.api.EncryptionKeyInfo;
+import org.apache.pulsar.client.api.Producer;
+import org.apache.pulsar.client.api.ProducerConfiguration;
+import org.apache.pulsar.client.api.PulsarClient;
+import org.apache.pulsar.client.api.PulsarClientException;
+
+public class SampleCryptoProducer {
+
+    public static void main(String[] args) throws PulsarClientException, InterruptedException, IOException {
+
+        class EncKeyReader implements CryptoKeyReader {
+
+            Map<String, EncryptionKeyInfo> encKeys = new HashMap<String, EncryptionKeyInfo>();
+
+            EncKeyReader(String keyName, byte[] pubKey, byte[] privKey) {
```

Comment:
Why do we need to use a hash-map if the constructor only allows for a single (public, private) key pair? 
Couldn't we just use something like: 

```java
byte[] privateKey;
byte[] publicKey;
```

Reply:
While the interface can be used with multiple keys, the implementation cannot. I think it's a bit confusing and if it's current purpose it's to hole a pair of key contents, we should do it in the most straightforward way.

Reply:
The implementation itself should be able to handle multiple keys in some cases. For example, if the message is consumed by different application and each of them do not prefer to share their private key, they can give their public key to producer and producer will encrypt the session key with each of the public key for every message published. Consumer will go through each of the key and attempt to retrieve its private key and proceed with decryption with the one it managed to get from the interface.

For this example, we can just implement a File based reader.